Fix FileView

scober opened the pull request:
FileView is implemented a little sloppily. This PR takes some steps to address that.

danielmawhirter commented on libquery/src/Table.cpp:
```diff
@@ -46,7 +46,7 @@ struct AssignJSONValue {
       // TIME* indicates a duration since midnight
       // format as <hour>:<minute>:<second>.<fraction>
       arrow_vendored::date::to_stream(stream, "%R:%S", TimeUnit(value));
-      ok_ = galois::QueryErrorCode::Success;
```
This one bit me more than once in the past week, had to fix in my branch, happy to see the disambiguation.

danielmawhirter reviewed (approved):
LGTM

Commit:
FileView: valid_->bound_

This is a purely aesthetic change, but it makes more sense for the
Bind() function to result in a bound object (as opposed to a valid
object).

Also move private members lower in the file to match the style of the
repo.

Commit:
FileView: Unbind() more thoroughly

Reset all class members instead of just a few. Also add a defensive
check to Bind() that clears the valididty vector before resizing it.

e-mcginnis reviewed (approved):
Amazing, this fixes the issue I was observing. Thanks!
